### PR TITLE
[SPARK-23565] [SS] New error message for structured streaming sources assertion

### PR DESCRIPTION
**Pull request by patrickmcgloin**
## What changes were proposed in this pull request?

A more informative message to tell you why a structured streaming query cannot continue if you have added more sources, than there are in the existing checkpoint offsets.

## How was this patch tested?

I added a Unit Test.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeq.scala [hunk 1]
```diff
@@ -39,7 +39,9 @@ case class OffsetSeq(offsets: Seq[Option[Offset]], metadata: Option[OffsetSeqMet
    * cannot be serialized).
    */
   def toStreamProgress(sources: Seq[BaseStreamingSource]): StreamProgress = {
-    assert(sources.size == offsets.size)
+    assert(sources.size == offsets.size, s"There are [${offsets.size}] sources in the " +
+      s"checkpoint offsets and now there are [${sources.size}] sources requested by the query.  " +
```
Comment (xuanyuanking) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeq.scala]: nit: extra blank in the end

File: sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLogSuite.scala [hunk 2]
```diff
@@ -125,6 +125,19 @@ class OffsetSeqLogSuite extends SparkFunSuite with SharedSQLContext {
     assert(offsetSeq.metadata === Some(OffsetSeqMetadata(0L, 1480981499528L)))
   }
 
+  test("assertion that number of checkpoint offsets match number of sources") {
```
Comment (xuanyuanking) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLogSuite.scala]: maybe no need add UT for the log change.

**Review by xuanyuanking — COMMENTED** [review]
Change the title to '[SPARK-23565][SS]...' is more accurate.

**Comment by patrickmcgloin** [comment]
Hi Xuan, 

I have updated the title, removed the extra white space after the period / dot and removed the UT.

**Comment by patrickmcgloin** [comment]
Hi @xuanyuanking, please see comments above.

**Review by xuanyuanking — APPROVED** [review]
+1 To me, useful in SS debugging, cc @cloud-fan to take a look.

**Comment by cloud-fan** [comment]
cc @zsxwing

**Comment by zsxwing** [comment]
okey to test

LGTM

**Comment by zsxwing** [comment]
ok to test

**Comment by zsxwing** [comment]
Thanks! Merging to master.

**Pull request closed without merging**